src/utils/transaction.ts

```diff
@@ -185,99 +160,59 @@ export const buildCrossChainAction = async (
-                  createTimestamp,
-                  status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
-                })
-              }
-              return transactions.push({
-                to: step.to as string,
-                value: step.value,
-                data: step.data,
-                createTimestamp,
-                status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
-              })
-            })
-          } else {
-            transactions = advancedRouteSteps.map(({
-              to,
-              value,
-              data,
-              chainId,
-            }) => ({
-              to: to as string,
-              value,
-              data,
-              chainId: chainId ?? fromChainId,
-              createTimestamp,
-              status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
-            }));
-          }
-
-          if (ethers.utils.isAddress(bestRoute.fromToken.address)
-            && !addressesEqual(bestRoute.fromToken.address, nativeAssetPerChainId[fromChainId].address)
-            && transactions.length === 1
-            && bestRoute.fromAmount) {
+          transactions = [{
+            to: routeToUSDC.transaction.to,
+            value: routeToUSDC.transaction.value as string,
+            data: routeToUSDC.transaction.data as string,
+            createTimestamp,
+            status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+          }];
+
+          if (ethers.utils.isAddress(fromAssetAddress)
+            && !addressesEqual(fromAssetAddress, nativeAssetPerChainId[fromChainId].address)
+            && routeToUSDC.approvalData?.approvalAddress) {
             const abi = getContractAbi(ContractNames.ERC20Token);
-            const erc20Contract = sdk.registerContract<ERC20TokenContract>('erc20Contract', abi, bestRoute.fromToken.address);
-            const approvalTransactionRequest = erc20Contract?.encodeApprove?.(transactions[0].to, bestRoute.fromAmount);
+            const erc20Contract = sdk.registerContract<ERC20TokenContract>('erc20Contract', abi, fromAssetAddress);
+            const approvalTransactionRequest = erc20Contract?.encodeApprove?.(routeToUSDC.approvalData.approvalAddress, routeToUSDC.approvalData.amount);
             if (!approvalTransactionRequest || !approvalTransactionRequest.to) {
               return { errorMessage: 'Failed build bridge approval transaction!' };
             }
 
             const approvalTransaction = {
               to: approvalTransactionRequest.to,
               data: approvalTransactionRequest.data,
-              value: 0,
+              value: '0',
               createTimestamp,
               status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
             };
 
             transactions = [approvalTransaction, ...transactions];
           }
 
-          const result = await klimaDaoStaking(BigNumber.from(bestRoute.toAmount).sub('250000').toString(), sdk.state.walletAddress, sdk);
+          const result = await klimaDaoStaking(routeToKlima, receiverAddress, sdk, true, '0');
```

Comment:
Please make sure we `.catch` here as the code below assumes a successful response with an error message

Reply:
it will since I have already put try blocks on the klimaDaoStaking function that it will return successful response either with an error message or with the data

Reply:
Got it cheers